app/screens/reactions/emoji_bar/item.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {TouchableOpacity, View} from 'react-native';
+import AnimatedNumbers from 'react-native-animated-numbers';
+
+import Emoji from '@components/emoji';
+import {useTheme} from '@context/theme';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+type ReactionProps = {
+    count: number;
+    emojiName: string;
+    highlight: boolean;
+    onPress: (emojiName: string) => void;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        count: {
+            color: changeOpacity(theme.centerChannelColor, 0.56),
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        countContainer: {marginRight: 5},
+        countHighlight: {
+            color: theme.buttonBg,
+        },
+        customEmojiStyle: {color: '#000'},
+        emoji: {marginHorizontal: 5},
+        highlight: {
+            backgroundColor: changeOpacity(theme.buttonBg, 0.08),
+        },
+        reaction: {
+            alignItems: 'center',
+            borderRadius: 4,
+            backgroundColor: theme.centerChannelBg,
+            flexDirection: 'row',
+            height: 32,
+            justifyContent: 'center',
+            marginRight: 12,
+            minWidth: 50,
+        },
+    };
+});
+
+const Reaction = ({count, emojiName, highlight, onPress}: ReactionProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const handlePress = useCallback(() => {
+        onPress(emojiName);
+    }, [highlight]);
+
+    return (
+        <TouchableOpacity
+            onPress={handlePress}
+            style={[styles.reaction, (highlight && styles.highlight)]}
+        >
+            <View style={styles.emoji}>
+                <Emoji
+                    emojiName={emojiName}
+                    size={20}
+                    textStyle={styles.customEmojiStyle}
+                    testID={`reaction.emoji.${emojiName}`}
+                />
+            </View>
+            <View style={styles.countContainer}>
+                <AnimatedNumbers
+                    includeComma={false}
+                    fontStyle={[styles.count, (highlight && styles.countHighlight)]}
```

Comment:
Could use a useMemo.

Reply:
it could, but is it worth it? I'm going to leave it like this until we do the profiling for everything.